Does not build on VC++ 2013

I am running Visual Studio 2013 with update 4 installed. This code has multiple build issues.

1) In the pair and tuple specializations for print_container_helper::printer, it chokes on this line:

```
using ostream_type = print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
```

I've fixed it locally with this:

```
#ifdef _MSC_VER
	using ostream_type = std::basic_ostream<TChar, TCharTraits>;
#else
	using ostream_type = print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
#endif
```

This isn't the greatest fix because it hardcodes it to basic_ostream, rather than referencing print_container_helper::ostream_type, but at least it works.

2) Once that builds, the library is unusable since it now chokes on the main interface that a caller would use: the << operator.

```
template<typename T, typename TChar, typename TCharTraits>
inline typename enable_if<::pretty_print::is_container<T>::value,
	   basic_ostream<TChar, TCharTraits> &>::type
	   operator<<(basic_ostream<TChar, TCharTraits>& stream, const T& container)
{
	return stream << ::pretty_print::print_container_helper<T, TChar, TCharTraits>(container);
}
```

Cannot resolve to the proper type in your most basic example:

```
  std::vector<int> foo;
  foo.push_back(1);
  foo.push_back(2);
  foo.push_back(3);

  std::cout << "My vector: " << foo << std::endl;
```

VS says:

```
error C2679: binary '<<' : no operator found which takes a right-hand operand of type 'std::vector<int,std::allocator<_Ty>>' (or there is no acceptable conversion)
```

The user must specifically add each type they want the overloaded << operator to work with, such as:

```c_cpp
template <typename T>
struct is_container<std::vector<T>> : std::true_type { };

template <typename T1, typename T2>
struct is_container<std::unordered_map<T1, T2>> : std::true_type { };
````

IMHO, if the main example shows it working with a vector, then it should work like that out of the box without the user having to specifically enter each type they want to work with.

With those changes, I have it working locally. Thanks.

--- a/prettyprint.hpp
+++ b/prettyprint.hpp
@@ -157,7 +157,7 @@ namespace pretty_print
     template <typename T1, typename T2>
     struct print_container_helper<T, TChar, TCharTraits, TDelimiters>::printer<std::pair<T1, T2>>
     {
-        using ostream_type = print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
+        using ostream_type = typename print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
 
         static void print_body(const std::pair<T1, T2> & c, ostream_type & stream)
         {
@@ -174,7 +174,7 @@ namespace pretty_print
     template <typename ...Args>
     struct print_container_helper<T, TChar, TCharTraits, TDelimiters>::printer<std::tuple<Args...>>
     {
-        using ostream_type = print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
+        using ostream_type = typename print_container_helper<T, TChar, TCharTraits, TDelimiters>::ostream_type;
         using element_type = std::tuple<Args...>;
 
         template <std::size_t I> struct Int { };

